lighthouse-core/audits/resource-summary.js

```diff
@@ -0,0 +1,119 @@
+  mediaResourceType: 'Media',
+  /** Label for a row in a data table; entries will be the total number and byte size of all 'Font' resources loaded by a web page */
+  fontResourceType: 'Font',
+  /** Label for a row in a data table; entries will be the total number and byte size of all resources loaded by a web page that don't fit into the categories of Document, Script, Stylesheet, Image, Media, & Font.*/
+  otherResourceType: 'Other',
+  /** Label for a row in a data table; entries will be the total number and byte size of all third-part resources loaded by a web page */
+  thirdPartyResourceType: 'Third-party',
+};
+
+const str_ = i18n.createMessageInstanceIdFn(__filename, UIStrings);
+
+class ResourceSummary extends Audit {
+  /**
+   * @return {LH.Audit.Meta}
+   */
+  static get meta() {
+    return {
+      id: 'resource-summary',
+      title: str_(UIStrings.title),
+      description: str_(UIStrings.description),
+      scoreDisplayMode: Audit.SCORING_MODES.INFORMATIVE,
+      requiredArtifacts: ['devtoolsLogs'],
+    };
+  }
+
+  /**
+   * @param {LH.Artifacts} artifacts
+   * @param {LH.Audit.Context} context
+   * @return {Promise<LH.Audit.Product>}
+   */
+  static async audit(artifacts, context) {
+    const devtoolsLog = artifacts.devtoolsLogs[Audit.DEFAULT_PASS];
+    const networkRecords = await NetworkRecords.request(devtoolsLog, context);
+    const mainResource = await (MainResource.request({devtoolsLog, URL: artifacts.URL}, context));
+    const resourceSummary = ComputedResourceSummary.summarize(networkRecords, mainResource.url);
+
+    /** @type {LH.Audit.Details.Table['headings']} */
+    const headings = [
+      {key: 'label', itemType: 'text', text: 'Resource Type'},
+      {key: 'count', itemType: 'numeric', text: 'Requests'},
+      {key: 'size', itemType: 'bytes', text: 'Transfer Size'},
+    ];
+
+
+    /** @type {Record<LH.Budget.ResourceType,string>} */
+    const strMappings = {
+      'total': str_(UIStrings.totalResourceType),
+      'document': str_(UIStrings.documentResourceType),
+      'script': str_(UIStrings.scriptResourceType),
+      'stylesheet': str_(UIStrings.stylesheetResourceType),
+      'image': str_(UIStrings.imageResourceType),
+      'media': str_(UIStrings.mediaResourceType),
+      'font': str_(UIStrings.fontResourceType),
+      'other': str_(UIStrings.otherResourceType),
+      'third-party': str_(UIStrings.thirdPartyResourceType),
+    };
+
+    const types = /** @type {Array<LH.Budget.ResourceType>} */ (Object.keys(resourceSummary));
+
+    const tableContents = types.map((type) => {
+      return {
+        // ResourceType is included as an "id". It does not appear directly in the table.
+        resourceType: type,
+        label: strMappings[type],
+        count: resourceSummary[type].count,
+        size: resourceSummary[type].size,
+      };
+    }).sort((a, b) => {
+      return b.size - a.size;
```

Comment:
is it weird to have `third-party` and `total` sorted in with the rest? I guess it'll always be `total` at the top, but then `third-party` also overlaps with the other categories (e.g. a third-party image is in both 'image' and 'third-party')...should we have `third-party` always at a fixed location in the table as well?

Reply:
What do you think of having third-party always be the last row in the table? It seems to me like the two logical places for it would either be as the second row or last row.

Reply:
> What do you think of having third-party always be the last row in the table?

Makes sense to me. It would be weird to have the second row be 0 and then high numbers after that for sites that are all first-party content, but I don't think cases where there is a high third-party number at the end would look out of place.